pkg/services/ngalert/api/api_provisioning.go

```diff
@@ -587,13 +588,18 @@ func exportResponse(c *contextmodel.ReqContext, body definitions.AlertingFileExp
 func exportHcl(download bool, body definitions.AlertingFileExport) response.Response {
 	resources := make([]hcl.Resource, 0, len(body.Groups)+len(body.ContactPoints)+len(body.Policies)+len(body.MuteTimings))
 	convertToResources := func() error {
-		for idx, group := range body.Groups {
+		for _, group := range body.Groups {
 			gr := group
-			resources = append(resources, hcl.Resource{
-				Type: "grafana_rule_group",
-				Name: fmt.Sprintf("rule_group_%04d", idx),
-				Body: &gr,
-			})
+			toHashedGruopIdx := append([]byte(gr.Name), []byte(gr.FolderUID)...)
+			hash, err := fnv.New32().Write(toHashedGruopIdx)
```

Comment:
NIT: Usually, we use 64-bit hash in code.

Reply:
fixed